### DISCUSSION TITLE
开发团队是否有多语言的计划？我可以提供支持

### DISCUSSION THREAD

**Discussion by orchief** [hhyo/Archery#2432] [Ideas]
我可以提供多语言的开发和相关支持，开发团队可以联系我

**Comment by LeoQuote** [comment]
欢迎pr，在实现时请尽量采用django 国际化模块进行实现，内部已有部分国际化代码，只是后续的开发没有太在意。

**Reply by Maxcj** [reply]
目前已经尝试使用 django 的国际化进行编写，但是有些问题，

Q1: 目前在仓库里面的代码页面显示都是中文，是否需要先将中文用英文短语进行替换，再使用{% trans "string" %}标签进行翻译？同时后面实现的功能是否也需要开发者提 PR 的时候一并带上多语言的翻译文本？

Q2:看到评论说内部已有部分国际化代码，这部分代码是？

**Reply by LeoQuote** [reply]
后端python 逻辑用django 的gettext https://github.com/hhyo/Archery/blob/c1d4e05451020a3212a9d626a4792cc7ca99cf6f/sql/models.py#L233

模板里我了解不多，应该是如你所说的。

js里是用类似 python 后端的 gettext 

可以多参考 django 的文档 https://docs.djangoproject.com/en/5.0/topics/i18n/translation/
